org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/editor/quickfix/QuickFixInjectorProvider.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.tests.editor.quickfix;
+
+import org.eclipse.xtext.testlanguages.fileAware.FileAwareTestLanguageRuntimeModule;
+import org.eclipse.xtext.testlanguages.fileAware.ide.FileAwareTestLanguageIdeModule;
+import org.eclipse.xtext.testlanguages.fileAware.ide.FileAwareTestLanguageIdeSetup;
+import org.eclipse.xtext.testlanguages.fileAware.ide.tests.FileAwareTestLanguageIdeInjectorProvider;
+import org.eclipse.xtext.util.Modules2;
+
+import com.google.inject.Binder;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+public class QuickFixInjectorProvider extends FileAwareTestLanguageIdeInjectorProvider {
+
+	// overridden to avoid NPE in org.eclipse.xtext.resource.impl.LiveShadowedResourceDescriptions.getAllResourceDescriptions()
+	@Override
+	public Injector internalCreateInjector() {
+		return new FileAwareTestLanguageIdeSetup() {
+			@Override
+			public Injector createInjector() {
+				return Guice.createInjector(Modules2.mixin(new FileAwareTestLanguageRuntimeModule(), new FileAwareTestLanguageIdeModule() {
+					@Override
+					public void configureIResourceDescriptionsLiveScope(Binder binder) {
+						// do not override runtime binding
```

Comment:
Why this

Reply:
The Unit test failed with an NPE because of the IDE implementation. I used the runtime binding and resolved it with it. I can attach a stacktrace of the failing test. If I can resolve it another way I'll try to.

Reply:
please do so

Reply:
Will push a commit that fixes the NPE

Reply:
i assume the problem is that you dont test on full lsp level and thus no ChunkedResourceDescriptions
are created. so you might have to change that or do something similar as others tests do

```
	protected def EObject createInMemoryFile(CharSequence content, EClass type) {
.....
// might also go to org.eclipse.xtext.ide.tests.serializer.ChangeSerializerTestHelper.createResourceSet(InMemoryURIHandler), dont know why it is not there
		val Map<String, ResourceDescriptionsData> dataMap = new HashMap();
		dataMap.put("egal", ResourceSetAdapter.findResourceDescriptionsData(rs));
		new ChunkedResourceDescriptions(Collections.unmodifiableMap(dataMap), rs); 
		return rs.contents(fileName, type.instanceClass) as EObject

```

this might also be a LiveShadowedChunkedResourceDescriptions vs LiveShadowedResourceDescriptions
problem (latter is used in normal changeserializer tests vs lsp/ide usually usss LiveShadowedChunkedResourceDescriptions and thus ChunkedResourceDescriptions to be installed (usually project and workspace manager care about this

Reply:
Thanks for the hints. I'll consider that fix.